File: src/main/java/org/example/keycloak/schemas/ScimPatchUserRequest.java
package org.example.keycloak.schemas;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.List;

@JsonIgnoreProperties(ignoreUnknown = true)
public record ScimPatchUserRequest(
    List<String> schemas,
    @JsonProperty("Operations")
    List<Operation> operations
) {

  @JsonIgnoreProperties(ignoreUnknown = true)
  public record Operation(
      String op,
      String path,
      String value
  ) {

  }
}

File: src/main/kotlin/org/example/keycloak/schemas/ScimPatchUserRequest.kt
package org.example.keycloak.schemas

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty

@JsonIgnoreProperties(ignoreUnknown = true)
data class ScimPatchUserRequest(
    val schemas: List<String>,
    @JsonProperty("Operations")
    val operations: List<Operation>
) {
    @JsonIgnoreProperties(ignoreUnknown = true)
    data class Operation(
        val op: String,
        val path: String,
        val value: String
    )
}